grammar/utils.js

```diff
@@ -403,6 +403,41 @@ class Utils {
         error.errorLocation = {line, column};
         return FormulaError.ERROR(msg, error);
     }
+    /*
+    * @param {*} multiArr : The Array that newArr's values will be put into
+    * @param {*} inputArr : either an array of arrays, or an array with no sub arrays
+    * 
+    * Each sub array of either array represents a new line in a spreadsheet.
+    * updateRow appends all of the inputArr's values to the end of multiArr sequentially,
+    * such that the columns of multiArr and inputArr align
+    *  
+    * updateRow([[1], [3]], [[2], [3]])
+    * becomes:
+    * multiRow = [[1,2], [3,4]]
+    * on the spread sheet:
+    * 1 2
+    * 3 4
+    */
+    updateRow = (multiArr, inputArr) => {
```

Comment:
Why is this code duplicated?

Reply:
The folders /grammar and /test/grammar are identical and both contain a utils file (where updateRow resides) and a hooks file (imports utils and parses). both copies of Hooks and utils are identical. I assumed one of the files was for testing purposes and the other for deployment. Thus I copied the function into both.

Reply:
TLDR, I removed updateRow from both independently and hit errors

Reply:
Put code in FormulaParser and import from there

Reply:
I did but depParser threw errors b/c it also calls the same parser. Thus, I added updateRow to it too. I tried to find a single file to put depParser into, but since the only values passed into Parser is DepParser/FormulaParser and their respective utils, I do not think it is possible to have updateRow in one place, unless you want to combine util files.

Reply:
I don't understand why the same parser impacts importing from a file, but let's just merge this in and follow up right after.